### PR TITLE
Disable saving and previewing while uploading

### LINKED ISSUE #2547: Story saved pointing to "blob" URL, causing video to fail loading in story
## Bug Description

For [this story](https://stories-new-wordpress-amp.pantheonsite.io/stories/bokunostory/) the background video in page one was saved using an incorrect blob url. Per @swissspidy 's review this is the markup: 
`<amp-video autoPlay="autoplay" poster="blob:https://stories-new-wordpress-amp.pantheonsite.io/13a6604b-2b51-4a6e-b4e6-7d0eceec4bf4" artwork="blob:https://stories-new-wordpress-amp.pantheonsite.io/13a6604b-2b51-4a6e-b4e6-7d0eceec4bf4" title="Bridge In Place" alt="Bridge In Place" layout="fill" id="el-5aff6cbd-9252-4e9d-a532-969cbde98f1a-media"><source type="video/mp4" src="blob:https://stories-new-wordpress-amp.pantheonsite.io/491491bd-b975-46b1-b337-5561e134bdce"/></amp-video>`

Also from Pascal: "See the blob: thing. That means the video & images were probably still uploading at the  time the story was saved, Of course a blob: URL isn’t really valid, hence the video fails to play"

## Expected Behaviour

Story should be saved with the proper video URL, no blobs.

## Steps to Reproduce

1. Drag a video onto canvas to upload while creating story
2. While video is uploading, save story/autosave? 

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by merapi**
## Summary

More info https://github.com/google/web-stories-wp/issues/2547

## Relevant Technical Choices

Had to move `AutoSaveHandler` below `MediaProvider` - checking the implications.

## User-facing changes

The user shouldn't be able to save/autosave the story while something is uploading (Autosave/Save/Preview buttons are disabled).

## Testing Instructions

Try to save/auto the story (open Preview tab and check if the story is updated there) while something is uploading - you shouldn't be able to do it.

---

Fixes #2547

### CODE REVIEW

File: assets/src/edit-story/components/autoSaveHandler/index.js [hunk 1]
```diff
@@ -40,22 +40,24 @@ function AutoSaveHandler() {
...
     let timeout = setTimeout(() => {
       saveStory();
-      timeout = null;
     }, autoSaveInterval * 1000);
 
     return () => {
-      timeout && clearTimeout(timeout);
+      clearTimeout(timeout);
     };
   }, [autoSaveInterval, isDraft, saveStory, status, hasNewChanges]);
```
Comment (swissspidy) [inline assets/src/edit-story/components/autoSaveHandler/index.js]:
```suggestion
  }, [autoSaveInterval, isDraft, saveStory, status, hasNewChanges, isUploading]);
```

File: assets/src/edit-story/components/header/buttons.js [hunk 2]
```diff
@@ -147,7 +150,7 @@ function PreviewButton() {
 
   return (
     <>
-      <Outline onClick={openPreviewLink} isDisabled={isSaving}>
+      <Outline onClick={openPreviewLink} isDisabled={isSaving || isUploading}>
```
Comment (dvoytenko) [inline assets/src/edit-story/components/header/buttons.js:153]: Do we need to do the same with "Save draft" button?

Comment (swissspidy) [reply assets/src/edit-story/components/header/buttons.js:153]: We already do that:

https://github.com/google/web-stories-wp/blob/7025fab42f299f2b30f3e9b06e4069396e02e997/assets/src/edit-story/components/header/buttons.js#L250-L257

**Review by swissspidy — APPROVED** [review]
Approved with 1  nit.

**Review by miina — APPROVED** [review]
Other than what Pascal already brought up LGTM

**Pull request merged**